From 621368b546dd2524cf544467eb18c70ffb3912d2 Mon Sep 17 00:00:00 2001
From: lmcneilx <lornax.mcneil@intel.com>
Date: Mon, 26 Mar 2018 16:43:25 +0100
Subject: [PATCH 93/96] ASoC: TDF8532: Allow time for I2C command to execute

Add a sufficient delay after i2c send / recv to allow for the I2C command
execution time. Delay chosen (2ms for send, 1ms for recv) is based on the
max time taken for the I2C command to execute, according to the TDF8532
user manual.

Signed-off-by: lmcneilx <lornax.mcneil@intel.com>
---
 sound/soc/codecs/tdf8532.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/soc/codecs/tdf8532.c b/sound/soc/codecs/tdf8532.c
index 938816c..067bad2 100644
--- a/sound/soc/codecs/tdf8532.c
+++ b/sound/soc/codecs/tdf8532.c
@@ -14,6 +14,7 @@
 #include <linux/i2c.h>
 #include <linux/jiffies.h>
 #include <linux/time.h>
+#include <linux/delay.h>
 #include <linux/acpi.h>
 #include <sound/soc.h>
 #include <sound/tlv.h>
@@ -70,6 +71,7 @@ static int __tdf8532_single_write(struct tdf8532_priv *dev_data,
 	if (ret < 0)
 		dev_err(dev, "i2c send packet returned: %d\n", ret);
 
+	mdelay(2);
 	dev_data->pkt_id++;
 
 	return ret;
@@ -131,6 +133,7 @@ static uint8_t tdf8532_single_read(struct tdf8532_priv *dev_data,
 		goto out_free;
 	}
 
+	mdelay(1);
 	return recv_len;
 
 out_free:
-- 
1.9.1

